keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/RoundVertex.java

```diff
@@ -16,16 +13,6 @@
      * @param inputVertex the vertex to be rounded
      */
     public RoundVertex(DoubleVertex inputVertex) {
-        super(inputVertex.getShape(), inputVertex);
-    }
-
-    @Override
-    protected DoubleTensor op(DoubleTensor a) {
-        return a.round();
-    }
-
-    @Override
-    public DualNumber calculateDualNumber(Map<Vertex, DualNumber> dualNumbers) {
-        throw new UnsupportedOperationException();
+        super(Tensor.SCALAR_SHAPE, inputVertex, DoubleTensor::round, null);
```

Comment:
This looks a bit rogue, why are we setting a scalar shape here?

I think you want: 
```
public RoundVertex(DoubleVertex inputVertex) {
        super(inputVertex, DoubleTensor::round);
    }
```

Reply:
thanks